snap/snapcraft.yaml

```diff
@@ -1,52 +1,105 @@
-name: spark-k8s # you probably want to 'snapcraft register <name>'
+name: spark-client
 base: core20
 version: '0.1'
-summary: Client side snap to submit spark jobs to kubernetes.
+summary: Client side scripts to submit Spark jobs to a cluster.
 description: |
-  Spark is now available on Kubernetes and needs a client to submit and 
-  interact with jobs. This snap packages the client binaries for the same.
+  The spark-client snap includes the scripts spark-submit, spark-shell, pyspark and other tools for managing Apache Spark jobs.
 
-grade: devel
-confinement: strict
+grade: stable
+confinement: classic
+
+plugs:
+  enable-scala-history:
+    interface: personal-files
+    write:
+    - $HOME/.scala_history
 
 apps:
+  setup-spark-k8s:
+    command: ops/setup_wrapper.sh
+    environment:
+      JAVA_HOME: $SNAP/usr/lib/jvm/java-11-openjdk-amd64
+      PATH: $JAVA_HOME/bin:$PATH
+      PYTHONPATH: $PYTHONPATH:$SNAP/usr/lib/python3/dist-packages
+      OPS_ROOT: ${SNAP}/ops
   spark-submit:
     command: bin/spark-submit
     environment:
       JAVA_HOME: $SNAP/usr/lib/jvm/java-11-openjdk-amd64
       PATH: $JAVA_HOME/bin:$PATH
+    plugs:
+        - network
+        - network-bind
+        - network-control
+        - home
   spark-shell:
     command: bin/spark-shell
     environment:
       JAVA_HOME: $SNAP/usr/lib/jvm/java-11-openjdk-amd64
       PATH: $JAVA_HOME/bin:$PATH
+    plugs:
+        - network
+        - network-bind
+        - network-control
+        - home
+        - enable-scala-history
   pyspark:
     command: bin/pyspark
     environment:
+      JAVA_HOME: $SNAP/usr/lib/jvm/java-11-openjdk-amd64
+      PATH: $JAVA_HOME/bin:$PATH
       PYTHONPATH: $PYTHONPATH:$SNAP/usr/lib/python3/dist-packages
+    plugs:
+        - network
+        - network-bind
+        - network-control
+        - home
 
 parts:
   spark:
-    source: https://github.com/apache/spark.git
     plugin: nil
-    build-packages: 
-        - git
-        - openjdk-11-jdk
-        - maven
+    build-packages:
+        - ca-certificates
+        - ca-certificates-java
+        - openjdk-11-jre-headless
         - python3
-        - python3-pip
         - wget
     stage-packages:
-        - openjdk-11-jre
+        - openjdk-11-jre-headless
+    stage-snaps:
+        - kubectl
```

Comment:
This is not working @grobbie . What would be the correct way to specify this uber snap dependency?

To keep the implementation functional for now I have kept the classic confinement to work with my locally installed microk8s.kubectl.